tests/handlers/test_user_directory.py

```diff
@@ -1045,7 +1047,9 @@ class TestUserDirSearchDisabled(unittest.HomeserverTestCase):
 
     def make_homeserver(self, reactor: MemoryReactor, clock: Clock) -> HomeServer:
         config = self.default_config()
-        config["update_user_directory"] = True
+        # Remove the value that disables updating the user directory, as that function
+        # is needed below. It will be force disabled later
+        config.pop("update_user_directory_from_worker")
```

Comment:
I'm not really following this one -- it seems like it would be clearer to set it to an explicit value or something?

Reply:
You're right. Not sure what I was thinking. This needs to be disabled in a different way and re-enabled by setting to `None` for the main process(although removal in this way does the same thing).